packages/rum/src/domain/getSessionReplayLink.ts

```diff
@@ -0,0 +1,50 @@
+import type { RumConfiguration, RumSessionManager, ViewContexts } from '@datadog/browser-rum-core'
+import { getDatadogOrigin } from '@datadog/browser-rum-core'
+import { isBrowserSupported } from '../boot/isBrowserSupported'
+import { getReplayStats } from './replayStats'
+
+export function getSessionReplayLink(
+  configuration: RumConfiguration,
+  sessionManager: RumSessionManager,
+  viewContexts: ViewContexts
+): string | undefined {
+  const session = sessionManager.findTrackedSession()
+  const queryParams: Record<string, string> = {}
+  let sessionId: string
+  if (session) {
+    sessionId = session.id
+
+    if (session.sessionReplayAllowed === false) {
+      // possibilities
+      // - replay sampled out
+      queryParams['error-type'] = 'incorrect-session-plan'
```

Comment:
<!-- Suggestions are specific requests to improve the subject under review. It is assumed that we all want to do what's best, so these comments are never dismissed as “mere suggestions”, but are taken seriously.  -->
**💬 suggestion:** ‏Having a separate function to get the error type would be clearer:

```ts
function getErrorType(session: RumSession | undefined, view: ViewContext | undefined) {
  if (!isBrowserSupported()) {
    return 'browser-not-supported'
  }
  if (!session) {
    // possibilities:
    // - rum sampled out
    // - session expired (edge case)
    return 'rum-not-tracked'
  }
  if (!session.sessionReplayAllowed) {
    // possibilities
    // - replay sampled out
    return 'incorrect-session-plan'
  }
  if (view && !getReplayStats(view.id)) {
    return 'replay-not-started'
  }
}
```